src/libs/Navigation/CustomActions.js

```diff
@@ -6,40 +6,27 @@ import lodashGet from 'lodash/get';
 import linkingConfig from './linkingConfig';
 import navigationRef from './navigationRef';
 
+/**
+ * @returns {Object}
+ */
+function getActiveState() {
+    // We use our RootState as the dispatch's state is relative to the active navigator and might not contain our active screen.
+    return navigationRef.current.getRootState();
+}
+
 /**
  * Go back to the Main Drawer
  * @param {Object} navigationRef
  */
 function navigateBackToRootDrawer() {
-    let isLeavingNestedDrawerNavigator = false;
-
-    // This should take us to the first view of the modal's stack navigator
-    navigationRef.current.dispatch((state) => {
-        // If this is a nested drawer navigator then we pop the screen and
-        // prevent calling goBack() as it's default behavior is to toggle open the active drawer
-        if (state.type === 'drawer') {
-            isLeavingNestedDrawerNavigator = true;
-            return StackActions.pop();
-        }
-
-        // If there are multiple routes then we can pop back to the first route
-        if (state.routes.length > 1) {
-            return StackActions.popToTop();
-        }
+    const activeState = getActiveState();
 
-        // Otherwise, we are already on the last page of a modal so just do nothing here as goBack() will navigate us
-        // back to the screen we were on before we opened the modal.
-        return StackActions.pop(0);
+    //Dispatch "popToTop" action takes you back to the first screen in the stack at the same time 
+    //it will prevent bubbling of action and limit it to activeState by specifing target.
```

Comment:
```suggestion
    // To navigate to the main drawer Route, pop to the first route on the Root Stack Navigator as the main drawer is always the first route that is activated.
    // It will pop all fullscreen and RHN modals that are over the main drawer. 
   // It won't work when the main drawer is not the first route of the Root Stack Navigator which is not the case ATM.
```